ecosystem/typescript/sdk_v2/src/types/index.ts

```diff
@@ -817,3 +820,48 @@ export enum AuthenticationKeyScheme {
    */
   DeriveResourceAccountAddress = 255,
 }
+
+/**
+ * TRANSACTION BUILDER
+ */
+
+export type TransactionData = {
+  secondary_signer_addresses?: Array<HexInput>;
+  fee_payer_address?: HexInput;
+} & (EntryFunctionData | ScriptData | EntryFunctionBCSData);
+
+export type TransactionPayload = {
+  type: "entry_function_payload" | "script_payload";
+  payload: TransactionPayloadEntryFunction | TransactionPayloadScript;
+};
+
+export type AptosRawTransaction = {
```

Comment:
I think we should either keep this `RawTransactionWithData` or at least clarify what exactly it is- because it isn't a `RawTransaction`.

If I understand it correctly, the extra data in `RawTransactionWithData`, i.e. the `secondary_signer_addresses` and `fee_payer_address` are only used in the transaction authentication process (signing), i.e. constructing the `TransactionAuthenticator`

They aren't included in the `RawTransaction` itself, so this name is a little misleading.

Not sure how much it matters, but just saying!

Also, fwiw, the struct in Rust is also named `RawTransactionWithData`.

The clearest name would probably be something like `RawTransactionWithAuthenticatorData` or something 😅